definitions/checks/foreman/validate_external_db_version.rb

```diff
@@ -0,0 +1,26 @@
+module Checks
+  module Foreman
+    class ValidateExternalDbVersion < ForemanMaintain::Check
+      metadata do
+        description 'Make sure server is running on required database version'
+        tags :pre_upgrade
+        label :validate_external_db_version
+        confine do
+          feature(:foreman_database) && !feature(:foreman_database).local? &&
+            !check_min_version('foreman', '2.0')
+        end
+      end
+
+      def run
+        current_db_version = feature(:foreman_database).db_version
+        fail!(db_upgrade_message(current_db_version)) if current_db_version.major < 12
+      end
+
+      def db_upgrade_message(db_version)
+        "\n\n*** WARNING: Server is running on PostgreSQL #{db_version}.\n"\
+        "*** Before performing the upgrade, please upgrade your database to PostgreSQL (>=12)\n"\
+        "*** otherwise data will be lost.\n"
```

Comment:
Does this mean we still support old pgsql and repercussion is data loss only ? If we are not going to support the old version then message should be modified.

Reply:
@upadhyeammit, @ehelms, any suggestion on this warning message?

Reply:
Yea, this is not a skill-able step. So I think it's ok for our language to be a bit stronger. The user *must* upgrade to Postgres 12 before continuing the upgrade.

Reply:
Re-phrased the warning message. Adding console output for reference:

```
================================================================================
Make sure server is running on required database version:             [FAIL]


*** WARNING: Server is running on PostgreSQL 9.2.24 DB.
*** Targetted Satellite does not support PostgreSQL versions less than 12.
*** Before proceeding further, make sure you must upgrade database to PostgreSQL (>=12).

--------------------------------------------------------------------------------
```

Reply:
>  Adding console output for reference:

```
================================================================================
Make sure server is running on required database version:             [FAIL]


*** WARNING: Server is running on PostgreSQL 9.2.24 DB.
*** Newer version of Satellite supports only PostgreSQL version 12 or above.
*** Before proceeding further, make sure you must upgrade database to PostgreSQL (>=12).

--------------------------------------------------------------------------------
```